### PR TITLE
Nova same client

### PR DESCRIPTION
**Pull request by seankelly**
Use the same client for all interactions with Nova. This allows not having to store the auth details in the class and caching the connection to OpenStack across that worker's life cycle.

### CODE REVIEW

File: master/docs/relnotes/index.rst [hunk 1]
```diff
@@ -28,6 +28,8 @@ Features
 * new :bb:reporter:`HipchatStatusPush` to report build results to Hipchat.
 * new steps for Visual Studio 2015 (VS2015, VC14, and MsBuild14).
 
+* :bb:worker:OpenStackLatentWorker now uses a single novaclient instance to not require re-authentication when starting or stopping instances.
```
Comment (rutsky) [inline master/docs/relnotes/index.rst]: You are missing backticks here:

```
:bb:worker:`OpenStackLatentWorker`
```

**Comment by rutsky** [comment]
@seankelly can you describe changes in docs indexes that you done? I don't understand them.

Btw, here is built version of docs from this PR: https://circle-artifacts.com/gh/buildbot/buildbot/765/artifacts/0/home/ubuntu/buildbot/master/docs/_build/html/index.html

**Comment by seankelly** [comment]
The [CircleCI results](https://circleci.com/gh/buildbot/buildbot/763) had an error:

    Warning, treated as error:
    /home/ubuntu/buildbot/master/docs/relnotes/index.rst:31: ERROR: Unknown interpreted text role "bb:worker".

That led me to try to figure out what that meant. The `bbdocs/ext.py` file extends what Sphinx will recognize to support `bb:worker`. There is already a `bb` domain that supports a few roles like `steps` and `reporter`. I added a `worker` role. I then annotated the other latent workers to bring them to the same level as OpenStackLatentWorker.

The change on the worker page can be seen by comparing the previous build (https://circle-artifacts.com/gh/buildbot/buildbot/764/artifacts/0/home/ubuntu/buildbot/master/docs/_build/html/manual/cfg-workers-openstack.html#worker-OpenStackLatentWorker) to this build. There is now a reference to where the class is located.

**Comment by rutsky** [comment]
@seankelly sounds reasonable, thanks for the explanation.

Can't say anything about OpenStack changes, but docs changes LGTM.

**Comment by seankelly** [comment]
I will rebase this once #2257 is merged to fix the conflict.

**Comment by seankelly** [comment]
Rebased.

**Pull request merged**